Examples/Tests/ElectrostaticSphere/analysis_electrostatic_sphere.py

```diff
@@ -0,0 +1,115 @@
+# r(0) = r_0, r'(0) = 0, and a = q_e*q_tot/(4*pi*eps_0*m_e)
+#
+# Note: E might be saved on a staggered time grid shifted by dt/2 (check this!)
+v_exact = lambda r: np.sqrt((q_e*q_tot)/(2*pi*m_e*eps_0)*(1/r_0-1/r))
+t_exact = lambda r: np.sqrt(r_0**3*2*pi*m_e*eps_0/(q_e*q_tot)) \
+    * (np.sqrt(r/r_0-1)*np.sqrt(r/r_0) \
+       + np.log(np.sqrt(r/r_0-1)+np.sqrt(r/r_0)))
+func = lambda rho: t_exact(rho)-t_max+dt/2  #Objective function to find r(t_max)
+r_end = fsolve(func,r_0)[0]  #Numerically solve for r(t_max)
+E_exact = lambda r: np.sign(r)*(q_tot/(4*pi*eps_0*r**2)*(abs(r)>=r_end) \
+    + q_tot*abs(r)/(4*pi*eps_0*r_end**3)*(abs(r)<r_end))
+
+# Open plotfile specified in command line (commented lines for debugging)
+# if __name__ == "__main__":
+#     filename = 'diags/plotfiles/plt00030'
+# else:
+#     filename = sys.argv[1]
+filename = sys.argv[1]
+ds = yt.load( filename )
+
+# Load data pertaining to fields
+data = ds.covering_grid(level=0, left_edge=ds.domain_left_edge,
+                        dims=ds.domain_dimensions)
+Ex = data['Ex'].to_ndarray()
+Ey = data['Ey'].to_ndarray()
+Ez = data['Ez'].to_ndarray()
+nx = Ex.shape[0];  ny = Ey.shape[1];  nz = Ez.shape[2]
+
+# Extract longitudinal fields along x-, y-, and z-axes
+Ex_axis = Ex[:,int(ny/2),int(nz/2)]
+Ey_axis = Ey[int(nx/2),:,int(nz/2)]
+Ez_axis = Ez[int(nx/2),int(ny/2),:]
+
+# Compute cell centers for grid
+x_cell_edges = np.linspace(xmin,xmax,nx+1)
+x_cell_centers = (x_cell_edges[1::]+x_cell_edges[0:-1])/2
+y_cell_edges = np.linspace(ymin,ymax,ny+1)
+y_cell_centers = (y_cell_edges[1::]+y_cell_edges[0:-1])/2
+z_cell_edges = np.linspace(zmin,zmax,nz+1)
+z_cell_centers = (z_cell_edges[1::]+z_cell_edges[0:-1])/2
+
+# Extract subgrid away from boundary (exact solution assumes infinite/open
+# domain but WarpX solution assumes perfect conducting walls)
+x_sub_grid = x_cell_centers[int(nx/4):int(3*nx/4)]
+y_sub_grid = y_cell_centers[int(ny/4):int(3*ny/4)]
+z_sub_grid = z_cell_centers[int(nz/4):int(3*nz/4)]
+
+# Exact solution of field along Cartesian axes
+Ex_exact_grid = E_exact(x_sub_grid)
+Ey_exact_grid = E_exact(y_sub_grid)
+Ez_exact_grid = E_exact(z_sub_grid)
+
+# WarpX solution of field along Cartesian axes
+Ex_grid = Ex_axis[int(nx/4):int(3*nx/4)]
+Ey_grid = Ey_axis[int(ny/4):int(3*nz/4)]
+Ez_grid = Ez_axis[int(ny/4):int(3*nz/4)]
+
+# Define approximate L2 norm error between exact and numerical solutions
+L2_error_x = np.sqrt(sum((Ex_exact_grid-Ex_grid)**2)) \
+    / np.sqrt(sum((Ex_exact_grid)**2))
+L2_error_y = np.sqrt(sum((Ey_exact_grid-Ey_grid)**2)) \
+    / np.sqrt(sum((Ey_exact_grid)**2))
+L2_error_z = np.sqrt(sum((Ez_exact_grid-Ez_grid)**2)) \
+    / np.sqrt(sum((Ez_exact_grid)**2))
+
+print("L2 error along x-axis = %s" %L2_error_x)
+print("L2 error along y-axis = %s" %L2_error_y)
+print("L2 error along z-axis = %s" %L2_error_z)
+
+assert L2_error_x < 0.05
+assert L2_error_y < 0.05
+assert L2_error_z < 0.05
```

Comment:
Files should end in a single, empty newline (makes diffs more readable)
```suggestion
assert L2_error_z < 0.05

```